requirements.txt

```diff
@@ -5,3 +5,7 @@ docutils >= 0.17.1
 
 # For RSS
 feedgen >= 0.9.0  # For RSS feed
+
+# For tests
+pytest
+pytest-cov
```

Comment:
This may slow down the main CI job a little as it will have to install `pytest` et al -- could we add a requirements file within `tests/`?

Reply:
We could but I doubt there'll be any noticeable difference, they're quick to download/install, and we're caching the pip directory inside GitHub Actions anyway.

Reply:
Comparing [a run of this PR](https://github.com/python/peps/runs/6109022947?check_suite_focus=true) with [a run of another](https://github.com/python/peps/runs/6109031282?check_suite_focus=true), the `Installing dependencies` step only takes 9s for the former instead of 7s for the latter, which is pretty minor, but I noticed the dependencies are installed a second time in the environment created by `make pages` creating and using the venv created with `make venv`, which takes 1:30 instead of 1:17, meaning the full run takes 1:46 instead of 1:31.

That's not the end of the world, but not entirely trivial either, and the dependencies installed the first time are much faster, but aren't actually used directly at all. The most obvious solution would be to just not do the first dep install at all, which AFAIK doesn't actually do anything but cache the dependencies for the second one. This ensures the logic in our Makefile is still tested, but does appear to be a fair bit less performant, especially with Pytest involved for some reason. To greatly speed up the build (and use more rigorous options), we _could_ run `python -b -X dev build.py -n -j 8 --build-dirs` directly, though in that case we wouldn't be using a `venv`, which isn't ideal.

Reply:
Good point about installing `requirements.txt` twice on the CI.

We can remove `python -m pip install --upgrade -r requirements.txt` from the yml, and only install via `make venv`.

I believe pip should still be cached by GHA. The pip _cache_ is still in the system location, whether it's installing into a venv or not.

---

I don't think those timings are a fair comparison, there's some level of variation anyway.

The last 5 `main` "Render PEPs", showing total time, install and render steps:

* 1m52 8s 1m31 https://github.com/python/peps/runs/6103232034?check_suite_focus=true
* 1m40 8s 1m16 https://github.com/python/peps/runs/6092797036?check_suite_focus=true
* 1m45 6s 1m21 https://github.com/python/peps/runs/6092777207?check_suite_focus=true
* 1m48 7s 1m15 https://github.com/python/peps/runs/6086747971?check_suite_focus=true
* 2m01 9s 1m33 https://github.com/python/peps/runs/6074335413?check_suite_focus=true

And the four on this branch:

* 1m33 9s 1m17 https://github.com/python/peps/runs/6095958355?check_suite_focus=true
* 1m46 9s 1m30 https://github.com/python/peps/runs/6109022947?check_suite_focus=true
* 2m02 10s 1m25 https://github.com/python/peps/runs/6112034078?check_suite_focus=true
* 1m36 7s 1m20 https://github.com/python/peps/runs/6112053388?check_suite_focus=true